### PR TITLE
Fix Apply SyntheticGroup

### PR DESCRIPTION
**Pull request by MasseGuillaume**
ex: (new A())() would become new A()()

### CODE REVIEW

File: tests/slow/coverage.txt [hunk 1]
```diff
@@ -1,13 +1,7 @@
 platform/yggdrasil/src/main/scala/com/precog/yggdrasil/table/CPathTraversal.scala
-sbt/main/src/main/scala/sbt/ConsoleProject.scala
```
Comment (MasseGuillaume) [inline tests/slow/coverage.txt]: 7!

File: tests/unit/src/test/scala/org/scalafmt/tests/TermSuite.scala [hunk 2]
```diff
@@ -46,8 +46,8 @@ object TermSuite extends BaseScalaPrinterTest {
           |  (<b></b>)
           |}
         """.stripMargin)
-  check("(a + b) { c => d }")
-  check("(a + b) { case c => d }")
+  check("((a + b)) { c => d }")
```
Comment (olafurpg) [inline tests/unit/src/test/scala/org/scalafmt/tests/TermSuite.scala]: Why are these parentheses needed? The trees are the same
```scala
ollie-scala@ q"(a + b) { _ + 1 }"
res1: Term.Apply = Term.Apply(
  Term.ApplyInfix(Term.Name("a"), Term.Name("+"), List(), List(Term.Name("b"))),
  List(Term.Block(List(Term.ApplyInfix(_, Term.Name("+"), List(), List(Lit.Int(1))))))
)

ollie-scala@ q"((a + b)) { _ + 1 }"
res2: Term.Apply = Term.Apply(
  Term.ApplyInfix(Term.Name("a"), Term.Name("+"), List(), List(Term.Name("b"))),
  List(Term.Block(List(Term.ApplyInfix(_, Term.Name("+"), List(), List(Lit.Int(1))))))
)
```

**Review by olafurpg — CHANGES_REQUESTED** [review]

File: tests/unit/src/test/scala/org/scalafmt/tests/TermSuite.scala (hunk 2 above)
Comment (MasseGuillaume) [reply tests/unit/src/test/scala/org/scalafmt/tests/TermSuite.scala]: check also does the diff, Do you prefer if we use `checkStructural` here ?

Comment (olafurpg) [reply tests/unit/src/test/scala/org/scalafmt/tests/TermSuite.scala]: It seems our precedence groups could be stricter to avoid that parenthesis.

Comment (olafurpg) [reply tests/unit/src/test/scala/org/scalafmt/tests/TermSuite.scala]: I think we should special case when Term.New is the qualifier in Term.Apply.

Comment (MasseGuillaume) [reply tests/unit/src/test/scala/org/scalafmt/tests/TermSuite.scala]: There was a miscommunication we cleared out offline: @olafurpg wants correctness but also does not want to add extra parenthesis when not necessary.

File: format/src/main/scala/org/scalafmt/internal/TreeDocOps.scala [hunk 3]
```diff
@@ -165,6 +165,10 @@ object TreeDocOps {
     SimpleExpr.wrap(lhs) + sep + rhs
   }
 
+  def dPath2(lhs: Tree, sep: Doc, rhs: Doc): Doc = {
```
Comment (olafurpg) [inline format/src/main/scala/org/scalafmt/internal/TreeDocOps.scala]: Can we come up with a more descriptive name here? Is this method necessary? Seems we might as well inline it.

File: format/src/main/scala/org/scalafmt/internal/TreeDocOps.scala [hunk 4]
```diff
@@ -165,6 +165,10 @@ object TreeDocOps {
...
+    print(lhs) + sep + rhs
```
Comment (olafurpg) [inline format/src/main/scala/org/scalafmt/internal/TreeDocOps.scala]: Don't we need to wrap Term.Apply qualifies in a syntactic group?

**Review by olafurpg — CHANGES_REQUESTED** [review]
Is `dPath` still necessary? The comment even says it's a "hack"

File: format/src/main/scala/org/scalafmt/internal/TreeDocOps.scala (hunk 4 above)
Comment (MasseGuillaume) [reply format/src/main/scala/org/scalafmt/internal/TreeDocOps.scala]: It's already wrapped in the only place it was called. Now, I completely removed it and it's more obvious.

File: tests/slow/coverage.txt [hunk 5]
```diff
@@ -1,10 +1,4 @@
...
-sbt/main/src/main/scala/sbt/Defaults.scala
-scala-js/library/src/main/scala/scala/scalajs/runtime/Bits.scala
 scala-js/test-suite/js/src/test/resources/SourceMapTestTemplate.scala
-scala/src/compiler/scala/tools/nsc/transform/Constructors.scala
-scala/src/compiler/scala/tools/nsc/transform/SpecializeTypes.scala
 scala/test/files/jvm/javaReflection/Classes_1.scala
 scala/test/files/pos/t9050.scala
 scala/test/files/run/t2378.scala
```
Comment (MasseGuillaume) [inline tests/slow/coverage.txt]: 4 to go!

Comment (MasseGuillaume) [inline tests/slow/coverage.txt:4]: https://github.com/scalameta/scalameta/issues/1385

File: tests/slow/coverage.txt [hunk 6]
```diff
@@ -1,10 +1,4 @@
-sbt/main/src/main/scala/sbt/ConsoleProject.scala
-sbt/main/src/main/scala/sbt/Defaults.scala
-scala-js/library/src/main/scala/scala/scalajs/runtime/Bits.scala
 scala-js/test-suite/js/src/test/resources/SourceMapTestTemplate.scala
```
Comment (MasseGuillaume) [inline tests/slow/coverage.txt:1]: https://github.com/scalameta/scalameta/issues/1384

File: tests/slow/coverage.txt [hunk 7]
```diff
@@ -1,10 +1,4 @@
...
-scala/src/compiler/scala/tools/nsc/transform/Constructors.scala
-scala/src/compiler/scala/tools/nsc/transform/SpecializeTypes.scala
 scala/test/files/jvm/javaReflection/Classes_1.scala
 scala/test/files/pos/t9050.scala
```
Comment (MasseGuillaume) [inline tests/slow/coverage.txt:3]: scala-syntax bug

File: tests/slow/coverage.txt [hunk 8]
```diff
@@ -1,10 +1,4 @@
-sbt/main/src/main/scala/sbt/ConsoleProject.scala
-sbt/main/src/main/scala/sbt/Defaults.scala
-scala-js/library/src/main/scala/scala/scalajs/runtime/Bits.scala
 scala-js/test-suite/js/src/test/resources/SourceMapTestTemplate.scala
-scala/src/compiler/scala/tools/nsc/transform/Constructors.scala
-scala/src/compiler/scala/tools/nsc/transform/SpecializeTypes.scala
 scala/test/files/jvm/javaReflection/Classes_1.scala
```
Comment (MasseGuillaume) [inline tests/slow/coverage.txt:2]: scala-syntax bug

**Review by olafurpg — APPROVED** [review]
That's more like it! 👍 👍 

> 4 left

woot 😸

**Pull request merged**